librz/core/cmd_debug.c

```diff
@@ -4219,6 +4219,37 @@ RZ_IPI int rz_debug_continue_oldhandler(void *data, const char *input) {
 	return 1;
 }
 
+RZ_IPI RzCmdStatus rz_cmd_debug_step_until_handler(RzCore *core, int argc, const char **argv) {
+	rz_reg_arena_swap(core->dbg->reg, true);
+	step_until(core, rz_num_math(core->num, argv[1]));
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_debug_step_until_instr_handler(RzCore *core, int argc, const char **argv) {
+	step_until_inst(core, argv[1], false);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_debug_step_until_instr_regex_handler(RzCore *core, int argc, const char **argv)  {
+	step_until_inst(core, argv[1], true);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_debug_step_until_optype_handler(RzCore *core, int argc, const char **argv) {
```

Comment:
I think this should be refactored and the command seems to accept a list of optypes... See inside `step_until_optype`, there is a `rz_str_split_list`. So you should add the `ARRAY` flag to the argument and refactor the `step_until_optype` function to actually accept a list of optypes instead of doing the split itself.

Reply:
So, should `step_until_optype` accept an `RzList` ?

Reply:
Yes

Reply:
Sorry for the late reply.
What should I use to convert `input`  to an `RzList` so that I can pass to it? That's where I was stuck.

Reply:
So, I have pushed a patch (breaking and probably incorrect) so that you can review a lot of other things as well.